presto-main/src/main/java/io/prestosql/execution/buffer/OutputBuffer.java

```diff
@@ -35,14 +36,14 @@
     boolean isFinished();
 
     /**
-     * Get the memory utilization percentage.
+     * Get the memory utilization ratio.
```

Comment:
what is the utilization of finished buffer?

Reply:
for a finished task's buffer it comes to be 0 - I think it should be 0 in other cases well like abort/destroy buffer since it leads to a full deref of the pages - and with that the mem. utilized by them is also released.

Reply:
Technically, it probably should be `empty()` since there is no longer a buffer to read from (`utilization=0/0`)

Reply:
currently a `FINISHED` buffer would very quickly lead to a finished task as well. Although, I can set it as `empty()` if you think it looks better - it makes sense to keep it `empty()` ideally.

Reply:
But 1 catch I see doing that is - we can return the `bufferState` in TaskStatus to be `FINISHED` but the `bufferOutputUtilization` to be non-empty and vice-versa since we fetch the state and util separately. Actually that is possible for other cases as well due to async updates to output buffer.

Reply:
Do we rely on `0` utilization for finished buffers somewhere (like scaled writers)?

Reply:
No, we do not. I think we should consider un-initialized and finished buffers in similar manner at co-ordinator conceptually.